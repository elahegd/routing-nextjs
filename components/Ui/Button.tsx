import React, { FC } from "react";
import Link from "next/link";

import classes from "./Button.module.css";

interface ButtonProps {
  link?: string;
  children: React.ReactNode;
  onClick?: () => void;
}

const Button: FC<ButtonProps> = ({ link, children, onClick }) => {
  if (link) {
    return (
      <Link className={classes.btn} href={link}>
        {children}
      </Link>
    );
  }

  return (
    <button onClick={onClick} className={classes.btn}>
      {children}
    </button>
  );
};

export default Button;
