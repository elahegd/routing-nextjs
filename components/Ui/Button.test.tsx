import React from "react";
import { render } from "@testing-library/react";

import Button from "./Button";

describe("Render and check props", () => {
  it("If button has link and children props", () => {
    const { asFragment } = render(<Button link="/">hey</Button>);

    expect(asFragment()).toMatchSnapshot();
  });

  it("If button has onClick and children props", () => {
    const handleClick = jest.fn();
    const { asFragment } = render(<Button onClick={handleClick}>hey</Button>);

    expect(asFragment()).toMatchSnapshot();
  });
});
