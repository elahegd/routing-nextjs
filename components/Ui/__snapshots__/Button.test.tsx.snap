// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render and check props If button has link and children props 1`] = `
<DocumentFragment>
  <a
    class="btn"
    href="/"
  >
    hey
  </a>
</DocumentFragment>
`;

exports[`Render and check props If button has onClick and children props 1`] = `
<DocumentFragment>
  <button
    class="btn"
  >
    hey
  </button>
</DocumentFragment>
`;
